### PR TITLE
perf: Optimize PullFeedAccountData aggregation and add unit tests

### PR DESCRIPTION
**Pull request by raushan728**
## Summary
Optimized price aggregation in `PullFeedAccountData` for better CU efficiency and memory usage.

## Key Changes
- **Memory:** Replaced `Vec` with stack-allocated array in `get_value`.
- **Logic:** Single-pass iteration in `current_result_ts_range` to save instructions.
- **Safety:** Used `saturating_sub` for slot arithmetic.
- **Tests:** Added coverage for median and range calculations.

### CODE REVIEW

File: src/on_demand/accounts/pull_feed.rs [hunk 1]
```diff
@@ -494,3 +533,36 @@ pub fn lower_bound_median(numbers: &mut [i128]) -> Option<i128> {
...
+        // test get_value median logic
+        let price = feed.get_value(100, 10, 3, true).unwrap();
+        // Now mantissa will correctly be 30 * 10^18
+        assert_eq!(price.mantissa(), 30 * scale_factor);
+
+        // test timestamp range logic
+        let (min_ts, max_ts) = feed.current_result_ts_range();
+        assert_eq!(min_ts, 1000);
+        assert_eq!(max_ts, 5000);
+    }
```
[the remark this answers is not in the document]
Comment (raushan728) [reply src/on_demand/accounts/pull_feed.rs:564]: I have expanded the unit test suite to 8 comprehensive cases. The new tests now cover staleness filtering, insufficient samples, empty submissions, and boundary cases for both even and odd submission counts.

File: src/on_demand/accounts/pull_feed.rs [hunk 2]
```diff
@@ -373,18 +373,35 @@ impl PullFeedAccountData {
...
         }
-        let median =
-            lower_bound_median(&mut submissions.iter().map(|s| s.value).collect::<Vec<_>>())
-                .ok_or(OnDemandError::NotEnoughSamples)?;
+
+        // in-place sort on the stack slice
+        let active_values = &mut values[..count];
+        active_values.sort_unstable();
+
+        let median = active_values[count / 2];
```
[the remark this answers is not in the document]
Comment (raushan728) [reply src/on_demand/accounts/pull_feed.rs]: Acknowledged. I have updated the median calculation to `(count - 1) / 2` to strictly implement the lower-bound pattern as suggested. This ensures logic parity with the legacy implementation for even-sized datasets.

**Comment by raushan728** [comment]
### PR Update: Review Feedback Addressed

I have updated the PR to incorporate the suggestions from the automated review:

- **Median Logic**: Refined the calculation to `(count - 1) / 2` to ensure strict lower-bound median selection.
- **Test Expansion**: Replaced the minimal test with a comprehensive suite of 8 unit tests covering staleness, insufficient samples, and edge cases for even/odd submission counts.

**Comment by raushan728** [comment]
@gallynaut @tamlut-modnys @jessupjn @mgild The review has been pending for a long time — please take a look once
